src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -1629,10 +1628,10 @@ private void configureEditProjectMetadataAction() {
   }
 
   private void configureShareThisPage() {
-    boolean showShare = !(entityBundle.getEntity() instanceof Project);
-    actionMenu.setActionVisible(Action.SHARE_THIS_PAGE, showShare);
-
-    if (showShare) {
+    if (hasProjectAreaContext(entityBundle.getEntity(), currentArea)) {
+      actionMenu.setActionVisible(Action.SHARE_THIS_PAGE, false);
+    } else {
+      actionMenu.setActionVisible(Action.SHARE_THIS_PAGE, true);
```

Comment:
Maybe a explicit comment for this case, letting the reader know that you are targeting both the top level Project Tools Menu and target (non-Project) entity Tools Menu